code/_onclick/ventcrawl.dm

```diff
@@ -45,77 +47,88 @@ var/list/ventcrawl_machinery = list(/obj/machinery/atmospherics/unary/vent_pump,
-		vent_found = clicked_on
+
+	var/obj/machinery/atmospherics/components/unary/vent_found
+
+
+	if(A)
+		vent_found = A
 		if(!istype(vent_found) || !vent_found.can_crawl_through())
 			vent_found = null
+
 	if(!vent_found)
-		var/obj/machinery/atmospherics/P
-		var/obj/O
-		for(O in range(1, src))
-			P = O
-			if(is_type_in_list(P, ventcrawl_machinery) && P.can_crawl_through())
-				vent_found = P
+		for(var/obj/machinery/atmospherics/machine in range(1,src))
+			if(is_type_in_typecache(machine, GLOB.ventcrawl_machinery))
+				vent_found = machine
+
+			if(!vent_found.can_crawl_through())
+				vent_found = null
+
+			if(vent_found)
 				break
-	if(!vent_found)
-		to_chat(src, "<span class='warning'>You must be standing on or beside an air vent to enter it.</span>")
-		return
-	if(!vent_found.network || !vent_found.network.normal_members.len && !vent_found.network.line_members.len)
-		to_chat(src, "<span class='warning'>This vent is not connected to anything.</span>")
-		return
-	if(!issilicon(src))
-		switch(vent_found.temperature)
-			if(0 to BODYTEMP_COLD_DAMAGE_LIMIT)
-				to_chat(src, "<span class='danger'>You feel a painful freeze coming from the vent!</span>")
-			if(BODYTEMP_COLD_DAMAGE_LIMIT to T0C)
-				to_chat(src, "<span class='warning'>You feel an icy chill coming from the vent.</span>")
-			if(T0C + 40 to BODYTEMP_HEAT_DAMAGE_LIMIT)
-				to_chat(src, "<span class='warning'>You feel a hot wash coming from the vent.</span>")
-			if(BODYTEMP_HEAT_DAMAGE_LIMIT to INFINITY)
-				to_chat(src, "<span class='danger'>You feel a searing heat coming from the vent!</span>")
-	visible_message("<span class='notice'>[src] begins climbing into [vent_found].</span>", \
-	"<span class='notice'>You begin climbing into [vent_found].</span>")
-	if(!do_after(src, 45, FALSE, 5, BUSY_ICON_GENERIC))
-		return
-	updatehealth()
-	if(stat || stunned || knocked_down || lying || health < 0)
-		return
-	if(!client)
-		return
-	if(!ventcrawl_carry())
-		return
-	visible_message("<span class='danger'>[src] scrambles into [vent_found]!</span>", \
-	"<span class='warning'>You climb into [vent_found].</span>")
-	if(!istype(src,/mob/living/carbon/Xenomorph/Hunter)) //Hunters silently enter/exit vents.
-		pick(playsound(src, 'sound/effects/alien_ventpass1.ogg', 35, 1), playsound(src, 'sound/effects/alien_ventpass2.ogg', 35, 1))
-	forceMove(vent_found)
-	add_ventcrawl(vent_found)
-	return
 
 
+	if(vent_found)
+		var/datum/pipeline/vent_found_parent = vent_found.parents[1]
+		if(vent_found_parent && (vent_found_parent.members.len || vent_found_parent.other_atmosmch))
+			visible_message("<span class='notice'>[src] begins climbing into the ventilation system...</span>" ,"<span class='notice'>You begin climbing into the ventilation system...</span>")
+
+			if(!do_after(src, 45, FALSE, 5, BUSY_ICON_GENERIC))
+				return
+
+			if(!client)
+				return
+
+			if(iscarbon(src) && can_ventcrawl())//It must have atleast been 1 to get this far
```

Comment:
```suggestion
			if(can_ventcrawl())//It must have atleast been 1 to get this far
```
The carbon check shouldn't be necessary, as it's covered in the other one.

Reply:
no because simple_animals arent carbon